packager/media/formats/mp4/mp4_muxer.cc

```diff
@@ -429,6 +430,36 @@ bool MP4Muxer::GenerateVideoTrak(const VideoStreamInfo* video_info,
   VideoSampleEntry video;
   video.format =
       CodecToFourCC(video_info->codec(), video_info->h26x_stream_format());
+  if (video.format == FOURCC_av01) {
+    auto av1_color_info =
+        base::SplitString(video_info->codec_string(), ".",
+                          base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+    if (av1_color_info.size() == 10) {
+      // Extract colr atom info if the full AV1 codec string is available.
+      // https://aomediacodec.github.io/av1-isobmff/#codecsparam
+      //   <sample entry 4CC>.<profile>.<level><tier>.<bitDepth>.<monochrome>.
+      //   <chromaSubsampling>.<colorPrimaries>.<transferCharacteristics>.
+      //   <matrixCoefficients>.<videoFullRangeFlag>
+      unsigned int color_primaries;
+      unsigned int transfer_characteristics;
+      unsigned int matrix_coefficients;
+      unsigned int video_full_range_flag;
+      video.colr.color_parameter_type = FOURCC_nclx;
```

Comment:
@cosmin, Is there an issue if we always write the colr atom as `nclx`? Even if the original atom was type `nclc`, we can make the output `nclx` by including the `video_full_range_flag` in the bitstream (it's value would be 0). Other than the inclusion of the `video_full_range_flag` data, is there any other differences between  `nclc` and `nclx`?

Reply:
From what I can tell for MP4 we always want to write `nclx`. The use of `nclc` should be just for reading from QuickTime MOV.